src/NuGet.Core/NuGet.DependencyResolver.Core/Remote/RemoteWalkContext.cs

```diff
@@ -49,5 +52,30 @@ public RemoteWalkContext(SourceCacheContext cacheContext, ILogger logger)
         /// True if this is a csproj or similar project. Xproj should be false.
         /// </summary>
         public bool IsMsBuildBased { get; set; }
+
+        /// <summary>
+        /// Applies Package namespace filtering for a given package
+        /// </summary>
+        /// <param name="libraryRange"></param>
+        /// <returns> Returns a subset of sources when namespaces are configured otherwise returns all the sources</returns>
+        public IList<IRemoteDependencyProvider> FilterDependencyProvidersForLibrary(LibraryRange libraryRange)
+        {
+            if (libraryRange == default)
+                throw new ArgumentNullException(nameof(libraryRange));
+
+            // filter package namespaces if enabled            
+            if (PackageNamespaces?.AreNamespacesEnabled == true && libraryRange.TypeConstraintAllows(LibraryDependencyTarget.Package))
+            {
+                IReadOnlyList<string> sources = PackageNamespaces.GetConfiguredPackageSources(libraryRange.Name);
+
+                if (sources == null || sources.Count == 0)
+                    throw new Exception(string.Format(CultureInfo.CurrentCulture,
```

Comment:
Who catches this exception?

In the restore command, we tend to `log` with the collector logger which ensures we write an assets file with a clear error message.

Reply:
Thanks for the details Nikolche. I modified code in the latest commit to write an error message to the log.